cmd/nerdctl/main_linux.go

```diff
@@ -39,16 +39,15 @@ func appNeedsRootlessParentMain(cmd *cobra.Command, args []string) bool {
 	switch commands[1] {
 	// completion, login, logout: false, because it shouldn't require the daemon to be running
 	// apparmor: false, because it requires the initial mount namespace to access /sys/kernel/security
-	// cp: false, because it requires the initial mount namespace to inspect file owners
```

Comment:
Why was this changed?

Reply:
I modified `appNeedsRootlessParentMain` to return true for `nerdctl cp` and `nerdctl container cp`. This was necessary to access containerd.sock in rootless mode to [create containerd client](https://github.com/containerd/nerdctl/pull/2355/files#diff-a6cc11fee8c6a93b90ff0594ba0ad1550e9e95de1f7a5f544cd47b8751c82addR64). When it's running in rootless mode without this change, get `/run/containerd/containerd.sock": no such file or directory `. However this change does not preserve the uid/gid mapping while copying to container. Should we disable the copying behaviour for stopped/created containers in rootless mode and only support this feature in rootful.

Reply:
> this change does not preserve the uid/gid mapping while copying to container.

This is a regression.

> Should we disable the copying behaviour for stopped/created containers in rootless mode and only support this feature in rootful.

Yes, at least for now.

Reply:
@AkihiroSuda I updated the PR to 

- use rootless containerd sock based on [faq.md](https://github.com/containerd/nerdctl/blob/main/docs/faq.md#containerd-socket-address) in rootless mode.
-  Disabled copying behaviour for stopped/created containers in rootless mode and only support this feature in rootful(more info in PR description). 
-  The uid/gid mapping is preserved when files are copied. 

PTAL, thanks! I noticed that the docker-compatibility check fails although all the tests for container copy pass. Not sure if this is related to this change. PTAL, thanks for the review.